### PR TITLE
chore(Bonus Pagamenti Digitali): [IAC-82] Graphical review for cashback states

### PR DESCRIPTION
**Pull request by fabriziofff**
## Short description
This pr introduces some changes for the texts and graphic states for the cashback.

Grace period            |  closed period, no supercashback, no max amount | closed period, max amount, no supercashback
:-------------------------:|:-----------:|:-------------------:
|   | 

closed period with supercashback   |  trx screen 
:-------------------------:|:--------------------------:
|   

## List of changes proposed in this pull request
- Copy update
- Changed rule to enhance text when the user earned the max amount or the supercashback.

### CODE REVIEW

File: locales/en/index.yml [hunk 1]
```diff
@@ -2223,10 +2223,11 @@ bonus:
           currentPeriodUnlockBody: "Congratulations {{name}}! You have reached the threshold of {{transactions}} transactions and unlocked your Cashback!"
           currentPeriodMaxAmount: "Congratulations {{name}}! You have reached the maximum amount expected!"
           currentPeriodSuperCashback: "Right now you would be eligible for {{superCashbackAmount}} € Super Cashback! Keep up the good work to stay on the leaderboard."
-          gracePeriodBody: "This Cashback period ended on {{date}}. We are waiting to receive the latest transactions to calculate refunds. By {{endGracePeriodDate}} you will be able to see the final calculation."
-          closedPeriodOKBody: "Congratulations {{name}}! \nYou are entitled to € {{amount}} of Cashback"
-          closedPeriodMaxAmount: "it's the highest refund possible!"
-          moneyTransfer: "By {{date}} you will receive the transfer on the IBAN you entered."
+          gracePeriodBody: "This Cashback period ended on {{date}}. We are waiting to receive the latest transactions to calculate refunds. You will receive a message when the operation is completed. "
```
Comment (thisisjp) [inline locales/en/index.yml:2226]:
```suggestion
          gracePeriodBody: "This Cashback period ended on {{date}}. We are waiting to receive the latest transactions to calculate refunds. You will receive a message when the operation is completed."
```

File: locales/en/index.yml [hunk 2]
```diff
@@ -2223,10 +2223,11 @@ bonus:
...
+          closedPeriodOKBody: "Congratulations {{name}}! \nYou are entitled to a refund of € {{amount}}"
+          closedPeriodMaxAmount: ", the highest refund possible"
+          closedPeriodSuperCashback: ", of which € {{amount}} of Super Cashback"
```
Comment (thisisjp) [inline locales/en/index.yml:2229]:
```suggestion
          closedPeriodSuperCashback: ", including € {{amount}} of Super Cashback"
```

File: locales/en/index.yml [hunk 3]
```diff
@@ -2223,10 +2223,11 @@ bonus:
...
+          moneyTransfer: "You will receive a message when the transfer is made to the IBAN of your choice."
```
Comment (thisisjp) [inline locales/en/index.yml:2230]:
```suggestion
          moneyTransfer: "You will receive a message when the transfer will be performed on the IBAN you entered."
```

**Review by thisisjp — APPROVED** [review]

File: locales/it/index.yml [hunk 4]
```diff
@@ -2252,10 +2252,11 @@ bonus:
...
           currentPeriodMaxAmount: "Complimenti {{name}}! Hai raggiunto l’importo massimo previsto!"
           currentPeriodSuperCashback: "In questo momento avresti diritto al Super Cashback di {{superCashbackAmount}} €! Continua così per rimanere in classifica."
-          gracePeriodBody: "Questo periodo del Cashback si è chiuso il {{date}}. Stiamo aspettando di ricevere le ultime transazioni, per calcolare i rimborsi. Entro il {{endGracePeriodDate}} potrai vedere il calcolo definitivo."
-          closedPeriodOKBody: "Complimenti {{name}}! \nHai diritto a € {{amount}} di Cashback"
-          closedPeriodMaxAmount: "è il massimo del rimborso possibile!"
-          moneyTransfer: "Entro il {{date}} riceverai il bonifico sull’IBAN che ci hai indicato."
+          gracePeriodBody: "Questo periodo del Cashback si è chiuso il {{date}}. Stiamo aspettando di ricevere le ultime transazioni, per calcolare i rimborsi. Riceverai un messaggio quando l'operazione verrà completata."
+          closedPeriodOKBody: "Complimenti {{name}}! \nHai diritto ad un rimborso di € {{amount}}"
+          closedPeriodMaxAmount: ", il massimo del rimborso possibile"
```
Comment (Undermaken) [inline locales/it/index.yml:2257]: @thisisjp Could this comma be misleading?
Schermata 2021-06-23 alle 16 37 54

File: ts/features/bonus/bpd/screens/details/components/summary/textualSummary/ClosedTextualSummary.tsx [hunk 5]
```diff
@@ -84,14 +114,7 @@ const OK = (props: Props) => (
...
       )}
-      {/* If the max amount is reached, inform the user */}
-      {props.period.amount.totalCashback >= props.period.maxPeriodCashback
-        ? ": " +
-          I18n.t(
-            "bonus.bpd.details.components.transactionsCountOverview.closedPeriodMaxAmount"
-          )
-        : "!"}
-      {"\n"}
+      {enhanceOkText(props).getOrElse("") + "!\n"}
```
Comment (Undermaken) [inline ts/features/bonus/bpd/screens/details/components/summary/textualSummary/ClosedTextualSummary.tsx:117]:
```suggestion
      {enhanceOkText(props)
        .map(t => t + "!\n")
        .getOrElse("")}
```
Is the `none` case possibile? if yes could the suggestion prevent we display a single question point?

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: ts/features/bonus/bpd/screens/details/components/summary/textualSummary/ClosedTextualSummary.tsx (hunk 5 above)
Comment (fabriziofff) [reply ts/features/bonus/bpd/screens/details/components/summary/textualSummary/ClosedTextualSummary.tsx:117]: Yes, the `none` case is possible (the user earn a cashback amount < max) and we should have always the final `!\n`, also for the amount < max.

File: locales/it/index.yml (hunk 4 above)
Comment (thisisjp) [reply locales/it/index.yml:2257]: If so, then Italians should adopt dot as decimal separator to avoid any confusion :P

**Review by Undermaken — APPROVED** [review]

**Pull request merged**